docs/install/mac.mdx

```diff
@@ -24,20 +24,13 @@ After you've downloaded the archive, uncompress it and move it into your `/Appli
 
 Espanso can also be installed through [Homebrew](https://brew.sh/), a popular macOS package manager.
 
-Because Espanso is hosted outside the core homebrew repository, you'll first need to add the appropriate
-_tap_ by running the following command inside a terminal:
-
-```
-brew tap espanso/espanso
-```
-
-Then, you can install Espanso by running the following command:
```

Comment:
Regarding this change I'm a bit torn. Yes, on one hand Espanso is now available on the main Homebrew tap thanks to the community, but on the other, we have no control over it. So if we publish a new Espanso version, we'll have to wait for the community to update it in the repo to be available to others.

Also, there's a chance the main Homebrew tap won't be as secure, as I don't know if there's anything preventing a malicious user to submitting a new Espanso version there that links to a malware rather than the real Espanso.

What do you think? @smeech @AucaCoyan

Reply:
I really know nothing about macOS and wrote this in conjunction with Archigos on [Discord](https://discord.com/channels/884163483409731584/1200502073754656838/1200933268648906832), but I can understand your concerns if `brew` isn't as carefully curated as the `apt` I'm familiar with on Linux.
I suppose we could include both sets of instructions, so people happy with the main Homebrew can use that and those who want to be sure of the latest version can use your instructions?

Reply:
I don't use macOS, and I can't use `cask` in linux, but I believe the line should be

```
brew install --cask espanso
```
because `cask` are graphical applications. 
As for the main brew vs tap, look at this graphical interface for brew:
![](https://camo.githubusercontent.com/737ad40d2c2dc6484fd34cf6c45944f06bfda58ac9409e186e841ba851fe0c97/68747470733a2f2f692e696d6775722e636f6d2f444e43737563702e706e67)
[source](https://github.com/buresdv/Cork#main-window)
taps, is like down below, right?

I don't think we should shy away from being on the main repository of brew. Are we (this 3 people in this conversation) capable of submitting a PR without having a macOS system?
We can also have it on the install section on the README flagged as "community install", "unofficial", or not in the "recommended" section.

What do you think?

Reply:
I don't know. As the instructions have been up for a while I assume people have been using them (although they may just be downloading the .ZIP) so they probably work. Your suggestion might be an improvement, of course.
Keeping the brew repository up to date might be ideal, but it's additional work and it's possible @federico-terzi might prefer to keep things in-house.
We really need the input of people who actually use macOS.

Reply:
Honestly, if Espanso still offers their own for Brew and that one is always updated as opposed to whoever is updating the 'official' brew version... I'd use that...  When I checked the docs for @smeech I had already just downloaded the *.app version and didn't install via Brew but updated it to reflect their release. I can try to remove the copy I have installed and do it via Brew (with screenshots) a little later today to see what the best process may be. If you have any questions/suggestions, let me know on Discord. @ me or DM me. :)

Reply:
Don't bother to re-install it! Do it only if you prefer it that way. We are confident that it works, otherwise people would file an issue.
We can continue as we are now and make the move on the future. It's nice we are talking about the topic! We don't need to change everything just because we are able to

Reply:
Not a problem, I do prefer *.app when I can over brew and the like... I can take a peak at the wording of it to maybe help explain why using the custom tap may be better than the one on the main tap

Reply:
Are the instructions in the [preview](https://vercel.live/open-feedback/espanso-website-git-fork-smeech-main-federicoterzi.vercel.app?via=pr-comment-visit-preview-link&passThrough=1) correct for both brew routes?

Reply:
Just chiming in as I'm hoping to get involved with Espanso (and contributed a tiny amount years ago to Espanso + Homebrew stuff: https://github.com/espanso/espanso/issues/615) -- 

A few thoughts while my coffee kicks in:

- IMO, Homebrew is the expected / established standard for installing tools for power users on macOS
- the Homebrew formulae are often (mostly?) maintained by FOSS contributors and volunteers and not necessarily by the official source code maintainers of a project
- security concerns are valid -- someone could theoretically host a keylogger version of Espanso and submit it to homebrew
    - concerns aren't *that* valid though, as someone could do this for literally any homebrew package (chrome, password managers, coreutils), or someone could also do this outside of homebrew -- whether by typo squatting or some Punycode URL or just by hosting an "unofficial" download mirror
    - concerns here can be mitigated by providing official hashes of the download / build artifacts, which homebrew uses in its package definitions to help guard against tampering, MITM attacks, etc. -- it looks like Espanso already does this in the GitHub releases -- compare the sha256 [here](https://github.com/Homebrew/homebrew-cask/blob/b3be530430344f52cc032ba6f54cf826b57987d5/Casks/e/espanso.rb#L5) and [here](https://github.com/espanso/espanso/releases/download/v2.2.1/Espanso-Mac-M1.zip.sha256.txt)
    - once a homebrew formulae is established, the maintainers would likely notice a significant change in the source domain: https://github.com/Homebrew/homebrew-cask/blob/b3be530430344f52cc032ba6f54cf826b57987d5/Casks/e/espanso.rb#L8
- IMO, ideally the Espanso team would be *welcome* to contribute to homebrew as a convenient means of installation for many users, but not expected or compelled to do so -- AFAICT the Espanso team has no more authority or expertise in maintaining its homebrew formula than any other individual (in other words, a PR to Homebrew from Espanso maintainers would be treated no differently than a PR from anyone else)

Also, please take the above with a grain of salt -- I have been actively moving away from Homebrew (mostly to nix), in large part because the Homebrew team seems very burnt out (for good reason), and their responses to issues and contributions are often curt at best, sometimes bordering on hostile.

Not that my thoughts should carry any weight, but my recommendation would be to focus on providing first-party downloads (from the website / github releases) and allow third parties to maintain the homebrew formulae.

Reply:
Thank you n8henrie!

Reply:
Thank you. That sounds very sensible. Are you happy with the instructions in the [docs](https://espanso.org/docs/install/mac/#install-using-homebrew)?

Reply:
Sorry for the delayed response. I think the current instructions are fine. The only modifications I think should be considered is to have the direct download specify that it is the *recommended* way to install (since it's under the Espanso team's control), and one could perhaps mention that the homebrew path is unofficial / not under the Espanso team's control. 

It also might be nice to mention nix as another unofficial installation method (if I can ever get it working on aarch64-darwin again: https://github.com/NixOS/nixpkgs/issues/247162).

Reply:
Sorry again for long response time.

> By direct download, you mean the [App Bundle](https://espanso.org/docs/install/mac/#install-as-app-bundle) route?

Yes

> How about:

According to https://github.com/espanso/espanso/issues/1804 the custom tap is not needed (and is extra / unnecessary maintenance burden on the Espanso team), so I would recommend removing this from the instructions. I would consider truncating it to just the following:

> Espanso can also be installed through [Homebrew](https://brew.sh/), a popular macOS package manager.

Could consider adding some verbiage along the lines of "the homebrew formula is not an official product of the Espanso team; please consider installing via the app bundle if you encounter issues."

Reply:
The wording I put [in](https://espanso.org/docs/install/mac/#install-using-homebrew) reflected a discussion that the as we have no control over the Homebrew version, we should indicate that the latest version is available via the custom tap, IIRC.

Reply:
@smeech yes, I see that, I'm just trying to confirm that it's accurate (and desirable).

From the currently live instructions:

> you may prefer to first add the project tap by running the following command inside a terminal

From the issue I posted above (emphasis mine):

> According to the instructions here: https://espanso.org/docs/install/mac the set up over brew should be done over a custom tap. In meantime espanso became available over [brew cask](https://github.com/Homebrew/homebrew-cask/blob/287f1f338d503afc206c4a24fbf4e00aae06d327/Casks/e/espanso.rb) **so the custom tap is not needed anymore.**

If I look at the old Federico... / homebrew repo, it obviously looks like a homebrew formula: https://github.com/federico-terzi/homebrew-espanso

I don't see anything like that at espanso/espanso. Does that tap even exist?

If so, should it be retired (now that the homebrew cask, which installs the macOS app bundle, seems to be preferred)?

Reply:
Hello! sorry for the late reply
No, the espanso/espanso tap doesn't exist to my knowledge. I believe we stick with federicoterzi/espanso because of the signing key https://github.com/espanso/espanso/issues/1928

To round up the disscusion, do you see anything to be improved? I would like to move to a new issue so we can track it better 😄